tensorflow/compiler/xla/service/gpu/cudnn_conv_algorithm_picker.cc

```diff
@@ -271,10 +271,10 @@ StatusOr<AutotuneResult> CudnnConvAlgorithmPicker::PickBestAlgorithmNoCache(
 
   int64 rng_state = 0;
 
-  const auto initialize_buffer = [stream, &result_shape,
-                                  &rng_state](DeviceMemoryBase buffer) {
-    InitializeFloatBuffer(stream, result_shape.element_type(), &rng_state,
-                          buffer);
+  const auto initialize_buffer = [&stream, &rng_state](
+                                     DeviceMemoryBase buffer,
+                                     const Shape& buffer_shape) {
+    InitializeBuffer(stream, buffer_shape.element_type(), &rng_state, buffer);
```

Comment:
What changed here? Isn't buffer_shape.element_type() the same as result_shape.element_type() ?

Reply:
initialize_buffer has two call sites down here.  Before this change, both took result_shape.  Now, since one takes operand->shape() and the other takes result_shape, we need to move buffer_shape from from captures to arguments for the lambda.